### PR TITLE
Fix EmbeddingData.embedding type

### PR DESCRIPTION
**Pull request by tddschn**

### CODE REVIEW

File: llmspec/llmspec.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from typing import Union, Optional, Dict, List
 from datetime import datetime
 from enum import Enum
+from llmspec.types import EmbeddingValue
```
Comment (kemingy) [inline llmspec/llmspec.py]: self import should be the last one

Comment (kemingy) [reply llmspec/llmspec.py]: I added more lint to catch this. Rebase the main branch and re-run then CI.

**Comment by kemingy** [comment]
Please resolve the conflicts.

File: llmspec/llmspec.py [hunk 2]
```diff
@@ -221,19 +215,24 @@ class ChatResponse(LMResponse):
...
 
 
 class EmbeddingData(msgspec.Struct):
-    embedding: List[float]
+    # not using EmbeddingValue
+    # TypeError: Type unions may not contain more than
+    # one array-like (list, set, tuple) type - type
+    # `typing.Union[typing.List[float], typing.List[typing.List[float]]]`
+    # is not supported
+    embedding: List[Union[float, List[float]]]
```
Comment (kemingy) [inline llmspec/llmspec.py:232]: This is not the correct type we expect. Since it can be `[1.0, [2.0]]`.

I will create another issue to track this.

**Review by kemingy — DISMISSED** [review]

**Comment by gaocegege** [comment]
Is this ready to review? @tddschn

**Review by kemingy — APPROVED** [review]

**Pull request merged**